src/Common/src/TypeSystem/Common/FieldDesc.FieldLayout.cs

```diff
@@ -43,7 +43,7 @@ public virtual bool HasGCStaticBase
             get
             {
                 if (!FieldType.IsValueType)
-                    return true;
+                    return !FieldType.IsPointer;
```

Comment:
BTW: We have been historically using `ContainsPointers` to mean whether the object contains any pointers tracked by the GC. It is mildly confusing because of `Pointer` is unmanaged in the type system. Maybe we should rename it to `ContainsObjRefs`?